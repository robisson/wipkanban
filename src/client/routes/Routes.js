// @flow
import * as React from "react";
import { Route, Switch } from "react-router-dom";
import FormLoginContainer from "../components/Login/FormLoginContainer";
import CreateAccountContainer from "../components/User/CreateAccountContainer";
import AccountCreated from "../components/User/AccountCreated";
import HomeContainer from "../components/home/HomeContainer";
import requireAuth from "../components/requireAuth";

const Routes: () => React.Node = (): React.Node => (
  <Switch>
    <Route exact path="/" component={requireAuth(HomeContainer)} />
    <Route path="/login" component={FormLoginContainer} />
    <Route path="/create-account" component={CreateAccountContainer} />
    <Route path="/account-created" component={AccountCreated} />
  </Switch>
);

export default Routes;
