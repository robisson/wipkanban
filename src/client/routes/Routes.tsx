// @flow
import * as React from "react";
import { Route, Switch } from "react-router-dom";
import LoginContainer from "../components/Login/LoginContainer";
import SignupContainer from "../components/Signup/SignupContainer";
import AccountCreated from "../components/Signup/AccountCreated";
import HomeContainer from "../components/Home/HomeContainer";
import requireAuth from "../components/requireAuth";
import BoardContainer from "../components/Board/BoardContainer";
import Signup from "../components/Signup";

const Routes: React.StatelessComponent<any> = () => (
  <Switch>
    <Route exact path="/" component={requireAuth(HomeContainer)} />
    <Route path="/board/:id" component={requireAuth(BoardContainer)} />
    <Route path="/signup" component={Signup} />
    <Route path="/login" component={LoginContainer} />
    <Route path="/create-account" component={SignupContainer} />
    <Route path="/account-created" component={AccountCreated} />
  </Switch>
);

export default Routes;
