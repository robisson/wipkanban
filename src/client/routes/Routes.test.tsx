import React from "react";
import Routes from "./Routes";
import { createShallow } from "@material-ui/core/test-utils";
import { Switch } from "react-router-dom";
import { ShallowWrapper } from "enzyme";

describe("<Routes /> Component", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Routes />);
  });

  it("should render an component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a Switch component", () => {
    expect(wrapper.type()).toEqual(Switch);
  });
});
