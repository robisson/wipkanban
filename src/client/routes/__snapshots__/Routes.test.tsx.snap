// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> Component should render an component correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/board/:id"
  />
  <Route
    component={[Function]}
    path="/signup"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/create-account"
  />
  <Route
    component={[Function]}
    path="/account-created"
  />
</Switch>
`;
